File: core/src/com/jupiter/europa/scene2d/TextureRegionActor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.jupiter.europa.scene2d;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.scenes.scene2d.Actor;

/**
 *
 * @author Nathan Templon
 */
public class TextureRegionActor extends Actor {

    // Fields
    private TextureRegion region;


    // Initialization
    public TextureRegionActor() {
        this.region = new TextureRegion();
    }
    
    public TextureRegionActor(TextureRegion region) {
        this.region = region;
    }


    // Public Methods
    public void setRegion(TextureRegion region) {
        this.region = region;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        System.out.println("Drawing");
        batch.draw(this.region, this.getX(), this.getY());
    }

}

File: core/src/com/jupiter/europa/scene2d/TextureRegionActor.kt
/*
 * The MIT License
 *
 * Copyright 2015 Nathan Templon.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 *
 */

package com.jupiter.europa.scene2d

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.scenes.scene2d.Actor

/**

 * @author Nathan Templon
 */
public class TextureRegionActor : Actor {

    // Fields
    private var region: TextureRegion? = null


    // Initialization
    public constructor() {
        this.region = TextureRegion()
    }

    public constructor(region: TextureRegion) {
        this.region = region
    }


    // Public Methods
    public fun setRegion(region: TextureRegion) {
        this.region = region
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        batch.draw(this.region, this.getX(), this.getY())
    }

}
